front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/RunAsUserToPermissionsMigration.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2019 Schibsted Media Group. All rights reserved
+ */
+
+package com.netflix.spinnaker.front50.migrations;
+
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO;
+import com.netflix.spinnaker.front50.model.pipeline.Trigger;
+import com.netflix.spinnaker.front50.model.serviceaccount.ServiceAccount;
+import com.netflix.spinnaker.front50.model.serviceaccount.ServiceAccountDAO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+
+@Component
+@ConditionalOnProperty("migrations.migrateToManagedServiceAccounts")
+public class RunAsUserToPermissionsMigration implements Migration {
+
+  // Only valid until April 1, 2020
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2020, Month.APRIL, 1);
+
+  private static final Logger LOG = LoggerFactory.getLogger(RunAsUserToPermissionsMigration.class);
+  private static final String SERVICE_ACCOUNT_SUFFIX = "@managed-service-account";
+  private static final String RUN_AS_USER = "runAsUser";
+  private static final String ROLES = "roles";
+
+  private final PipelineDAO pipelineDAO;
+  private final ServiceAccountDAO serviceAccountDAO;
+  private Clock clock = Clock.systemDefaultZone();
+
+  @Autowired
+  public RunAsUserToPermissionsMigration(PipelineDAO pipelineDAO, ServiceAccountDAO serviceAccountDAO) {
+    this.pipelineDAO = pipelineDAO;
+    this.serviceAccountDAO = serviceAccountDAO;
+  }
+
+  @Override
+  public boolean isValid() {
+    return LocalDate.now(clock).isBefore(VALID_UNTIL);
```

Comment:
I'm not sure any time window is really needed for this migration. People will probably be migrating to Pipeline Permissions for a longer time than this, given that it is still an opt-in feature.

Reply:
Agreed. I think this is fine. We can increase this later if needed.